panel/io/cache.py

```diff
@@ -20,6 +20,8 @@
 
 import param
 
+from param._utils import iscoroutinefunction
```

Comment:
:)

Maybe the utils module of Param should be made public and its functions private, except those that you use in Panel/HoloViews? Doesn't have to happen now, just brainstorming.

Reply:
I'm using this and a few other functions quite routinely so I'd be in favor. That said, I'll probably switch to `from param.parameterized import iscoroutinefunction` for now like I've done elsewhere in the codebase.

Reply:
You're certainly the one in the best place to decide where you want to import these from and how 👍 

I'm definitely in favor of being more explicit in Param about its public API and documenting whatever is public. I got bitten by this recently when refactoring `__init__.py`, it turns out some package (Panel maybe, I don't remember) relied on `_is_number` being a top level function :D

https://github.com/holoviz/param/blob/31f71c304f81ab14aec52d96f92653fbf6a8cf8e/param/__init__.py#L70